### ISSUE TITLE
DeferredAssert hangs when querying an MBean for a crashed server

### ISSUE THREAD

**Issue by brianoliver** [brianoliver/migration#80]
I had a misconfiguration that resulted in a cluster crash:

Exception in thread "main" (Wrapped: Failed to start Service "Management" (ServiceState=SERVICE_STOPPED)) java.lang.RuntimeException: Join request was aborted
at com.tangosol.coherence.component.util.daemon.queueProcessor.Service.start(Service.CDB:38)
at com.tangosol.coherence.component.util.daemon.queueProcessor.service.Grid.start(Grid.CDB:6)
at com.tangosol.coherence.component.util.SafeService.startService(SafeService.CDB:53)
at com.tangosol.coherence.component.util.SafeService.ensureRunningService(SafeService.CDB:27)
...

There is a DeferredAssert that ensures the right size cluster before proceeding with testing:

DeferredAssert.assertThat(eventually(invoking(cluster).getClusterSize()), is(nClusterMembers));

This call never returns, resulting in a locked up JVM with the following:

"main" prio=5 tid=7fcd95801000 nid=0x10106c000 waiting on condition [10106a000]
java.lang.Thread.State: TIMED_WAITING (sleeping)
at java.lang.Thread.sleep(Native Method)
at java.lang.Thread.sleep(Thread.java:302)
at java.util.concurrent.TimeUnit.sleep(TimeUnit.java:328)
at com.oracle.tools.deferred.Ensured.get(Ensured.java:207)
at com.oracle.tools.runtime.java.AbstractJavaApplication.getMBeanAttribute(AbstractJavaApplication.java:272)
at com.oracle.tools.runtime.coherence.ClusterMember.getClusterSize(ClusterMember.java:160)
at com.oracle.tools.runtime.coherence.Cluster.getClusterSize(Cluster.java:69)
at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
at java.lang.reflect.Method.invoke(Method.java:597)
at com.oracle.tools.deferred.DeferredInvoke.invoke(DeferredInvoke.java:175)
at com.oracle.tools.deferred.DeferredInvoke.get(DeferredInvoke.java:121)
at com.oracle.tools.deferred.Ensured.get(Ensured.java:184)
at com.oracle.tools.deferred.DeferredMatch.get(DeferredMatch.java:128)
at com.oracle.tools.deferred.DeferredMatch.get(DeferredMatch.java:48)
at com.oracle.tools.deferred.Ensured.get(Ensured.java:184)
at com.oracle.tools.deferred.DeferredHelper.ensure(DeferredHelper.java:214)
at com.oracle.tools.deferred.DeferredAssert.assertThat(DeferredAssert.java:131)
at com.oracle.tools.deferred.DeferredAssert.assertThat(DeferredAssert.java:66)
at federation.TestRunner.ensureCluster(TestRunner.java:222)
at federation.TestRunner.run(TestRunner.java:95)
at federation.JournalEntryTests.testJournalEntry_2Servers(JournalEntryTests.java:88)

**Comment by brianoliver** [comment]
This issue was automatically migrated from JIRA ORACLETOOLS-2

**Comment by brianoliver** [comment]
Caused by DeferredMBeanAttribute ignoring IOExceptions that indicate the MBeanServer which once was working was now lost - causing very long retries. Now it will fail fast.

**Comment by brianoliver** [comment]
Marked as **fixed** by @Unassigned on Monday, June 3rd 2013, 4:40:04 pm
